File: commons/src/main/java/com/ultreon/commons/lang/LoggableProgress.java
package com.ultreon.commons.lang;

/**
 * @author Qboi
 * @since 1.0.0
 */
public class LoggableProgress extends Progress {
    private final InfoTransporter infoTransporter;

    public LoggableProgress(InfoTransporter infoTransporter, int progress, int max) {
        super(progress, max);
        this.infoTransporter = infoTransporter;
    }

    public LoggableProgress(InfoTransporter infoTransporter, int max) {
        super(max);
        this.infoTransporter = infoTransporter;
    }

    public void log(String text) {
        infoTransporter.log(text);
    }
}

File: src/main/kotlin/qtech/bubbles/common/LoggableProgress.kt
package qtech.bubbles.common

class LoggableProgress : Progress {
    private val infoTransporter: InfoTransporter

    constructor(infoTransporter: InfoTransporter, progress: Int, max: Int) : super(progress, max) {
        this.infoTransporter = infoTransporter
    }

    constructor(infoTransporter: InfoTransporter, max: Int) : super(max) {
        this.infoTransporter = infoTransporter
    }

    fun log(text: String) {
        infoTransporter.log(text)
    }
}